packages/react-liveness/src/components/FaceLivenessDetector/service/utils/getFaceMatchStateInLivenessOval.ts

```diff
@@ -0,0 +1,127 @@
+}
+
+/**
+ * Returns the state of the provided face with respect to the provided liveness oval.
+ */
+export function getFaceMatchStateInLivenessOval({
+  face,
+  ovalDetails,
+  initialFaceIntersection,
+  sessionInformation,
+  frameHeight,
+}: MatchStateInOvalParams): {
+  faceMatchState: FaceMatchState;
+  faceMatchPercentage: number;
+} {
+  let faceMatchState: FaceMatchState;
+  const challengeConfig =
+    sessionInformation?.Challenge?.FaceMovementAndLightChallenge
+      ?.ChallengeConfig;
+  if (
+    !challengeConfig ||
+    !challengeConfig.OvalIouThreshold ||
+    !challengeConfig.OvalIouHeightThreshold ||
+    !challengeConfig.OvalIouWidthThreshold ||
+    !challengeConfig.FaceIouHeightThreshold ||
+    !challengeConfig.FaceIouWidthThreshold
+  ) {
+    throw new Error(
+      'Challenge information not returned from session information.'
+    );
+  }
+
+  const {
+    OvalIouThreshold,
+    OvalIouHeightThreshold,
+    OvalIouWidthThreshold,
+    FaceIouHeightThreshold,
+    FaceIouWidthThreshold,
+  } = challengeConfig;
+
+  const faceBoundingBox: BoundingBox = generateBboxFromLandmarks(
+    face,
+    ovalDetails,
+    frameHeight
+  );
+  const minFaceX = faceBoundingBox.left;
+  const maxFaceX = faceBoundingBox.right;
+  const minFaceY = faceBoundingBox.top;
+  const maxFaceY = faceBoundingBox.bottom;
+
+  const { ovalBoundingBox, minOvalX, minOvalY, maxOvalX, maxOvalY } =
+    getOvalBoundingBox(ovalDetails);
+
+  const intersection = getIntersectionOverUnion(
+    faceBoundingBox,
+    ovalBoundingBox
+  );
+
+  const intersectionThreshold = OvalIouThreshold;
+  const ovalMatchWidthThreshold = ovalDetails.width * OvalIouWidthThreshold;
+  const ovalMatchHeightThreshold = ovalDetails.height * OvalIouHeightThreshold;
+  const faceDetectionWidthThreshold = ovalDetails.width * FaceIouWidthThreshold;
+  const faceDetectionHeightThreshold =
+    ovalDetails.height * FaceIouHeightThreshold;
+
+  /** From Science
+   * p=max(min(1,0.75∗(si−s0)/(st−s0)+0.25)),0)
+   */
+  const faceMatchPercentage =
+    Math.max(
+      Math.min(
+        RANGE_MAX,
+        (WEIGHT_MAX * (intersection - initialFaceIntersection)) /
+          (intersectionThreshold - initialFaceIntersection) +
+          WEIGHT_MIN
+      ),
+      RANGE_MIN
+    ) * 100;
+
+  const faceIsOutsideOvalToTheLeft = minOvalX > minFaceX && maxOvalX > maxFaceX;
+  const faceIsOutsideOvalToTheRight =
+    minFaceX > minOvalX && maxFaceX > maxOvalX;
+
+  const faceIsMatched =
+    intersection > intersectionThreshold &&
+    Math.abs(minOvalX - minFaceX) < ovalMatchWidthThreshold &&
+    Math.abs(maxOvalX - maxFaceX) < ovalMatchWidthThreshold &&
+    Math.abs(maxOvalY - maxFaceY) < ovalMatchHeightThreshold;
+
+  const faceIsMatchedClosely =
+    minOvalY - minFaceY > faceDetectionHeightThreshold ||
+    maxFaceY - maxOvalY > faceDetectionHeightThreshold ||
+    (minOvalX - minFaceX > faceDetectionWidthThreshold &&
+      maxFaceX - maxOvalX > faceDetectionWidthThreshold);
```

Comment:
Why do we use `Math.abs` for the calculations in determining `faceIsMatched`  but not here?

Reply:
My guess is because `isFaceMatchedClosely` used to be `TOO_CLOSE` so a positive value for `minOvalY - minFaceY` (eg) indicates that the top of the face is overlapping with the top of the oval whereas a negative value might mean it's too far. But for the MATCHED state it could be a little close or a little far as long as it's within a range. But actually the new criteria for `isFaceMatched` is just `intersection > intersectionThreshold;`